cfme/services/catalogs/catalog.py

```diff
@@ -1,115 +1,110 @@
+        return (
+            self.title.text == "Adding a new Catalog" and
+            self.catalogs.is_opened
+        )
 
-item_multiselect = web_ui.MultiSelect(
-    "//select[@id='available_fields']",
-    "//select[@id='selected_fields']",
-    "//div[@id='column_lists']//a[contains(@href, 'button=right')]/img",
-    "//div[@id='column_lists']//a[contains(@href, 'button=left')]/img")
 
-form = web_ui.Form(
-    fields=[('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('button_multiselect', item_multiselect),
-            ('add_button', form_buttons.add),
-            ('save_button', form_buttons.save),
-            ('cancel_button', form_buttons.cancel)])
+class EditCatalogView(CatalogForm):
 
-item_form = web_ui.Form(
-    fields=[('type_select', "//select[@id='st_prov_type']"),
-            ('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('display_checkbox', Input("display")),
-            ('add_button', form_buttons.add)])
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Editing Catalog{}".format(self.obj.name) and
+            self.catalogs.is_opened
+        )
 
-match_page = partial(match_location, controller='catalog', title='Catalogs')
 
+class DetailsCatalogView(ServicesCatalogView):
 
-def _all_catalogs_add_new(_):
-    if BZ(1213863).blocks:
-        sel.pytest.skip("Blocker on the bug 1213863")
-    catalog_tree('All Catalogs')
-    cfg_btn('Add a New Catalog')
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == 'Catalog "{}"'.format(self.obj.name) and
+            self.catalogs.is_opened
+        )
 
 
 class Catalog(Updateable, Pretty, Navigatable):
-    """Represents a Catalog"""
-    pretty_attrs = ['name', 'items']
 
     def __init__(self, name=None, description=None, items=None, appliance=None):
+        Navigatable.__init__(self, appliance)
         self.name = name
         self.description = description
         self.items = items
-        Navigatable.__init__(self, appliance=appliance)
-
-    def __str__(self):
-        return self.name
 
     def create(self):
-        navigate_to(self, 'Add')
-        # For element not found exception (To be removed)
-        sel.sleep(5)
-        web_ui.fill(form, {'name_text': self.name,
-                           'description_text': self.description,
-                           'button_multiselect': self.items},
-                    action=form.add_button)
-        flash_str = 'Catalog "{}" was saved'
-        flash.assert_success_message(flash_str.format(self.name))
+        view = navigate_to(self, 'Add')
+        view.fill({'name': self.name,
+                   'description': self.description})
+        view.add_button.click()
+        view.flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
```

Comment:
I have done assert flash message before assert view in Add method so that while creating duplicate catalog item I can catch the error and view does not conflict . Please see if its okay.